src/can.rs

```diff
@@ -163,6 +163,67 @@ pub mod config {
         }
     }
 
+    /// Interrupt allocation flags.
+    /// These flags can be used to specify which interrupt qualities the code calling esp_intr_alloc* needs.
+    #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+    pub enum IntrFlags {
+        // Accept a Level 1 interrupt vector (lowest priority)
+        Level1,
+        // Accept a Level 2 interrupt vector.
+        Level2,
+        // Accept a Level 3 interrupt vector.
+        Level3,
+        // Accept a Level 4 interrupt vector.
+        Level4,
+        // Accept a Level 5 interrupt vector.
+        Level5,
+        // Accept a Level 6 interrupt vector.
+        Level6,
+        // Accept a Level 7 interrupt vector (highest priority)
+        Nmi,
+        // Interrupt can be shared between ISRs.
+        Shared,
+        // Edge-triggered interrupt.
+        Edge,
+        // ISR can be called if cache is disabled.
+        Iram,
```

Comment:
when this option is used it needs SDKCONFIG option CONFIG_TWAI_ISR_IN_IRAM enabled, so maybe gate it or add an info.

Reply:
I have updated the comment here. I'm not sure if can.rs is a good place for those flags. They are not only related to TWAI peripherals for example [I2C](https://github.com/esp-rs/esp-idf-hal/blob/331f3392a44f9e4bef7935df4e7186376b89d2f6/src/i2c.rs#LL188C29-L188C29).

Reply:
As we discussed yesterday I have moved the flags to intterupt.rs which is definitely a better place for them.